### PR TITLE
Support for named and bodyless reactions in C++

### PR DESCRIPTION
**Pull request by cmnrd**

### CODE REVIEW

**Review by petervdonovan — APPROVED** [review]
It is remarkable how simple this is in comparison to the C target. LGTM!

**Review by lhstrh — APPROVED** [review]
Nice! I wish all feature-adding PRs were this simple!

File: test/Cpp/src/hello_bodyless_world.cc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include "HelloBodylessWorld/HelloBodylessWorld.hh"
+
+void HelloBodylessWorld::Inner::hello_body([[maybe_unused]] const reactor::StartupTrigger& startup) {
```
Comment (revol-xut) [inline test/Cpp/src/hello_bodyless_world.cc:3]: so its always reaction_name + _body ? is this _body necessary ?

**Review by revol-xut — APPROVED** [review]

**Comment by lhstrh** [comment]
Can we include this in v0.5.0?

File: test/Cpp/src/hello_bodyless_world.cc (hunk 1 above)
Comment (cmnrd) [reply test/Cpp/src/hello_bodyless_world.cc:3]: I don't think it is required, actually.  However, there will be C++ compile errors if a reactor uses e.g. a state variable and a reaction with the same name. We  don't have proper name conflict handling.

Comment (cmnrd) [reply test/Cpp/src/hello_bodyless_world.cc]: Opened https://github.com/lf-lang/lingua-franca/issues/1973

**Pull request merged**